File: CS/B137591/MyTextEdit.cs
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraEditors.Registrator;
using DevExpress.XtraEditors.ViewInfo;
using DevExpress.XtraEditors.Drawing;
using System.ComponentModel;
using System;
using System.Collections.Generic;
using System.Globalization;
using DevExpress.XtraEditors.Mask;
using DevExpress.Utils;

namespace DXSample {
    public class MyTextEdit : TextEdit {
        static MyTextEdit() { RepositoryItemMyTextEdit.RegisterMytextEdit(); }
        public MyTextEdit() : base() { }
        public override string EditorTypeName { get { return RepositoryItemMyTextEdit.MyTextEditName; } }
        [DesignerSerializationVisibility(DesignerSerializationVisibility.Content)]
        public new RepositoryItemMyTextEdit Properties { 
            get { return (RepositoryItemMyTextEdit)base.Properties; } 
        }
        protected override void OnHandleCreated(EventArgs e) {
            base.OnHandleCreated(e);
            if (EditValue == null)
                Properties.AssignDefaultValue(this);
        }
    }
    [UserRepositoryItem("RegisterMyTextEdit")]
    public class RepositoryItemMyTextEdit : RepositoryItemTextEdit {
        static RepositoryItemMyTextEdit() { RegisterMytextEdit(); }
        public RepositoryItemMyTextEdit() : base() {
            Mask.MaskType = MaskType.Numeric;
            Mask.EditMask = "n0";
            valueType = AllowedValueTypes.Int32;
            AllowNullInput = DefaultBoolean.True;
        }
        internal const string MyTextEditName = "MyTextEdit";
        public override string EditorTypeName { get { return MyTextEditName; } }
        private AllowedValueTypes valueType;
        [DefaultValue(AllowedValueTypes.Int32)]
        public AllowedValueTypes ValueType {
            get { return valueType; }
            set { valueType = value; }
        }
        public static void RegisterMytextEdit() {
            EditorRegistrationInfo.Default.Editors.Add(new EditorClassInfo(MyTextEditName, typeof(MyTextEdit),
                typeof(RepositoryItemMyTextEdit), typeof(TextEditViewInfo), new TextEditPainter(), true));
        }
        public override void Assign(RepositoryItem item) {
            BeginUpdate();
            try {
                base.Assign(item);
                RepositoryItemMyTextEdit source = item as RepositoryItemMyTextEdit;
                if (source == null) return;
                valueType = source.valueType;
            } finally { EndUpdate(); }
        }
        internal void AssignDefaultValue(MyTextEdit editor) {
            switch (valueType) {
                case AllowedValueTypes.Byte: editor.EditValue = (byte)0; break;
                case AllowedValueTypes.Int16: editor.EditValue = (short)0; break;
                case AllowedValueTypes.Int32: editor.EditValue = (int)0; break;
                case AllowedValueTypes.Int64: editor.EditValue = (long)0; break;
            }
        }
        public override BaseEdit CreateEditor() {
            MyTextEdit editor = (MyTextEdit)base.CreateEditor();
            AssignDefaultValue(editor);
            return editor;
        }
    }
    public enum AllowedValueTypes { Byte, Int16, Int32, Int64 };
}

File: VB/B137591/MyTextEdit.vb
Imports Microsoft.VisualBasic
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraEditors.Registrator
Imports DevExpress.XtraEditors.ViewInfo
Imports DevExpress.XtraEditors.Drawing
Imports System.ComponentModel
Imports System
Imports System.Collections.Generic
Imports System.Globalization
Imports DevExpress.XtraEditors.Mask
Imports DevExpress.Utils

Namespace DXSample
	Public Class MyTextEdit
		Inherits TextEdit
		Shared Sub New()
			RepositoryItemMyTextEdit.RegisterMytextEdit()
		End Sub
		Public Sub New()
			MyBase.New()
		End Sub
		Public Overrides ReadOnly Property EditorTypeName() As String
			Get
				Return RepositoryItemMyTextEdit.MyTextEditName
			End Get
		End Property
		<DesignerSerializationVisibility(DesignerSerializationVisibility.Content)> _
		Public Shadows ReadOnly Property Properties() As RepositoryItemMyTextEdit
			Get
				Return CType(MyBase.Properties, RepositoryItemMyTextEdit)
			End Get
		End Property
		Protected Overrides Sub OnHandleCreated(ByVal e As EventArgs)
			MyBase.OnHandleCreated(e)
			If EditValue Is Nothing Then
				Properties.AssignDefaultValue(Me)
			End If
		End Sub
	End Class
	<UserRepositoryItem("RegisterMyTextEdit")> _
	Public Class RepositoryItemMyTextEdit
		Inherits RepositoryItemTextEdit
		Shared Sub New()
			RegisterMytextEdit()
		End Sub
		Public Sub New()
			MyBase.New()
			Mask.MaskType = MaskType.Numeric
			Mask.EditMask = "n0"
			valueType_Renamed = AllowedValueTypes.Int32
			AllowNullInput = DefaultBoolean.True
		End Sub
		Friend Const MyTextEditName As String = "MyTextEdit"
		Public Overrides ReadOnly Property EditorTypeName() As String
			Get
				Return MyTextEditName
			End Get
		End Property
		Private valueType_Renamed As AllowedValueTypes
		<DefaultValue(AllowedValueTypes.Int32)> _
		Public Property ValueType() As AllowedValueTypes
			Get
				Return valueType_Renamed
			End Get
			Set(ByVal value As AllowedValueTypes)
				valueType_Renamed = value
			End Set
		End Property
		Public Shared Sub RegisterMytextEdit()
			EditorRegistrationInfo.Default.Editors.Add(New EditorClassInfo(MyTextEditName, GetType(MyTextEdit), GetType(RepositoryItemMyTextEdit), GetType(TextEditViewInfo), New TextEditPainter(), True))
		End Sub
		Public Overrides Sub Assign(ByVal item As RepositoryItem)
			BeginUpdate()
			Try
				MyBase.Assign(item)
				Dim source As RepositoryItemMyTextEdit = TryCast(item, RepositoryItemMyTextEdit)
				If source Is Nothing Then
					Return
				End If
				valueType_Renamed = source.valueType
			Finally
				EndUpdate()
			End Try
		End Sub
		Friend Sub AssignDefaultValue(ByVal editor As MyTextEdit)
			Select Case valueType_Renamed
				Case AllowedValueTypes.Byte
					editor.EditValue = CByte(0)
				Case AllowedValueTypes.Int16
					editor.EditValue = CShort(Fix(0))
				Case AllowedValueTypes.Int32
					editor.EditValue = CInt(Fix(0))
				Case AllowedValueTypes.Int64
					editor.EditValue = CLng(Fix(0))
			End Select
		End Sub
		Public Overrides Function CreateEditor() As BaseEdit
			Dim editor As MyTextEdit = CType(MyBase.CreateEditor(), MyTextEdit)
			AssignDefaultValue(editor)
			Return editor
		End Function
	End Class
	Public Enum AllowedValueTypes
		[Byte]
		Int16
		Int32
		Int64
	End Enum
End Namespace